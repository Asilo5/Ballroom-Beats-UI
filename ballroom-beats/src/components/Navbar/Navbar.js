import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { createStackNavigator } from 'react-navigation-stack';
import { withNavigation } from 'react-navigation';

class Navbar extends Component {

  render() {
      return (
          <View style={styles.container}>
            <Text>Navbar</Text>
          </View>
      )      
  }
};  

const AppNavigator = createStackNavigator({
    Navbar: {
      screen: Navbar,
    },
});
  
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'white',
        alignItems: 'center',
        justifyContent: 'center',
    }
});

export default withNavigation(Navbar);