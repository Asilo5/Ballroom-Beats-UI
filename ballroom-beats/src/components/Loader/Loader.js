import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { createStackNavigator } from 'react-navigation-stack';
import { withNavigation } from 'react-navigation';
import Icon from 'react-native-vector-icons/AntDesign';
import Navbar from '../Navbar/Navbar';

class Loader extends Component {

  render() {
      return (
          <View style={styles.container}>
            <Text style={styles.countDown}>....to play time</Text>
            <View style={styles.btnContainer}>
                <Icon.Button style={styles.button1} 
                    name="arrowleft" 
                    onPress={() => this.props.navigation.navigate('SongPick')} 
                    title="BACK">
                    <Text style={styles.back}>Back</Text>
                </Icon.Button>
                <Icon.Button style={styles.button2} 
                    name="arrowright" 
                    onPress={() => this.props.navigation.navigate('Game')} 
                    title="NEXT">
                    <Text style={styles.play}>Play</Text>
                </Icon.Button>
            </View>
            <Navbar />
          </View>
      )      
  }
} 

const AppNavigator = createStackNavigator({
    Loader: {
      screen: Loader,
    },
});
  
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'black',
        alignItems: 'center',
        justifyContent: 'center',
    },
    countDown: {
        fontSize: 40,
        color: 'white',
        marginBottom: 10,
    },
    btnContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        width: '60%',
    },
    button1: {
        backgroundColor: 'green',
        fontSize: 25,
    },
    play: {
        fontSize: 28,
        fontWeight: 'bold',
    },
    button2: {
        backgroundColor: 'green',
        fontSize: 25,
    },
    back: {
        fontSize: 28,
        fontWeight: 'bold',
    }
});

export default withNavigation(Loader);