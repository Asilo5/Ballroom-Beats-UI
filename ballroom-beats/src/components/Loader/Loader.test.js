import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../Loader/Loader';
import Navbar from '../Navbar/Navbar';

jest.mock('../Navbar/Navbar')

const mockProps = {
   navigation: {
     navigate: jest.fn(),
     getParam: jest.fn(),
   }
 };

describe('Loader', () => {

    Navbar.mockImplementation(() => {
      return null
    })

   it('should match snapshot', () => {
    const loader = renderer.create(<Loader navigation={mockProps.navigation} />).toJSON();
    expect(loader).toMatchSnapshot();

   });

});
