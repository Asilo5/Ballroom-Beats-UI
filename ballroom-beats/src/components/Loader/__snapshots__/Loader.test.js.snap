// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#666363",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 35,
        "fontWeight": "bold",
        "marginBottom": 10,
        "textShadowColor": "#000",
        "textShadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "textShadowRadius": 2,
      }
    }
  >
    GET READY! Make sure your screen is oriented correctly...
  </Text>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 5,
                  "justifyContent": "center",
                  "marginHorizontal": 2,
                },
                Object {
                  "height": 91,
                  "width": 80.5,
                },
                Object {
                  "backgroundColor": "#666363",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontVariant": Array [
                      "tabular-nums",
                    ],
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontSize": 35,
                  },
                  Object {
                    "color": "white",
                    "textShadowColor": "#000",
                    "textShadowOffset": Object {
                      "height": 2,
                      "width": 2,
                    },
                    "textShadowRadius": 2,
                  },
                ]
              }
            >
              05
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "marginVertical": 2,
              },
              Object {
                "fontSize": 19.444444444444443,
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          Seconds
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#007AFF",
          "borderRadius": 5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 8,
          },
          Object {
            "backgroundColor": "#007AFF",
            "borderRadius": 5,
          },
          Object {
            "backgroundColor": "#A9C344",
            "fontSize": 25,
          },
        ]
      }
      title="BACK"
    >
      <Text />
      <Text
        style={
          Object {
            "fontSize": 28,
            "fontWeight": "bold",
          }
        }
      >
        Back
      </Text>
    </View>
  </View>
</View>
`;
